### PR TITLE
Implement WebTestCase for route testing

### PR DESCRIPTION
**Pull request by synbot**
### Acceptance Criteria
1. Not QA testable. Tests should exist and make sense and pass.

### Tasks
- Port it over from REDACTED, cleaning it up in the process.
- Write some example tests for existing routes

### Additional Notes
- {{list additional notes here, remove if not applicable}}

### CODE REVIEW

**Comment by zpchavez** [comment]
:snowflake: :arrow_up:

File: src/Synapse/TestHelper/WebTestCase.php [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        defined('APPDIR') or define('APPDIR', realpath(WEBDIR.'/..'));
+        defined('DATADIR') or define('DATADIR', APPDIR.'/data');
+        defined('TMPDIR') or define('TMPDIR', '/tmp');
+        date_default_timezone_set('UTC');
+
+        $applicationInitializer = new Synapse\ApplicationInitializer;
+
+        $app = $applicationInitializer->initialize();
+
+        $defaultServices = new Synapse\Application\Services;
```
Comment (baohx2000) [inline src/Synapse/TestHelper/WebTestCase.php]: I'm thinking there is probably a reduced set of providers we can use...for example, we won't need a db for these things.

**Comment by baohx2000** [comment]
:interrobang: one thought for speedier tests, see comment

**Comment by zpchavez** [comment]
:star:

File: config/db.php [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?php
```
Comment (baohx2000) [inline config/db.php]: Is this file or the init.php file still necessary with the removal of the extra providers?

**Comment by baohx2000** [comment]
:interrobang: can remove the db config file

**Comment by zpchavez** [comment]
:star: Looks like I still need the `init.php`.

**Comment by baohx2000** [comment]
:+1:

**Pull request merged**